### PR TITLE
#1622: Use PickerInline

### PR DESCRIPTION
**Pull request by vchoy**
Addresses: https://github.com/redwoodjs/redwood/issues/1622

### CODE REVIEW

**Comment by Tobbe** [comment]
Thanks for getting started on this @vchoy!

I noticed there are more instances of `<ReactFilestack>` in the document. Do you think you could update those as well?

File: cookbook/File_Upload.md [hunk 1]
```diff
@@ -164,17 +164,14 @@ Clicking that actually launches the picker with all kinds of options, like picki
 
 ![Filestack picker](https://user-images.githubusercontent.com/300/82617240-51e26000-9b84-11ea-8aec-210b7a751e8c.png)
 
-There's no reason to make the user click that button, let's just show the picker on the page when it loads by adding a couple of [options](https://github.com/filestack/filestack-react#props). We'll need to create a container for it to live in, so we'll add a `<div>` and give it an `id` attribute that we'll tell `<ReactFilestack>` about. We'll also give the `<div>` a couple of styles so that the picker doesn't collapse to 0px tall:
+There's no reason to make the user click that button, let's just show the picker on the page when it loads by adding a couple of [pickerOptions](https://github.com/filestack/filestack-react#props). We'll need to create a container for it to live in, so we'll add a `<div>` and have it as a child of `<PickerInline>`. We'll also give the `<div>` a couple of styles so that the picker doesn't collapse to 0px tall:
```
Comment (Tobbe) [inline cookbook/File_Upload.md:167]: We don't actually add any `pickerOptions`, right? So I think you can reword this sentence.

File: cookbook/File_Upload.md [hunk 2]
```diff
@@ -164,17 +164,14 @@ Clicking that actually launches the picker with all kinds of options, like picki
...
 // web/src/components/ImageForm/ImageForm.js
 
-<ReactFilestack
-  apikey={process.env.REDWOOD_ENV_FILESTACK_API_KEY}
-  componentDisplayMode={{ type: 'immediate' }}
-  actionOptions={{ displayMode: 'inline', container: 'picker' }}
-/>
-<div id="picker" style={{ marginTop: '2rem', height: '20rem' }}></div>
+<PickerInline apikey={process.env.REDWOOD_ENV_FILESTACK_API_KEY}>
+  <div id="picker" style={{ marginTop: '2rem', height: '20rem' }}></div>
```
Comment (Tobbe) [inline cookbook/File_Upload.md:173]: Is the `id` and the styles needed anymore? What does it look like with and without the styles?

File: cookbook/File_Upload.md [hunk 3]
```diff
@@ -209,12 +206,11 @@ const ImageForm = (props) => {
 
   // form stuff...
 
-  <ReactFilestack
-    apikey={process.env.REDWOOD_ENV_FILESTACK_API_KEY}
-    onSuccess={onFileUpload}
-    componentDisplayMode={{ type: 'immediate' }}
-    actionOptions={{ displayMode: 'inline', container: 'picker' }}
-  />
+	<PickerInline
```
Comment (Tobbe) [inline cookbook/File_Upload.md:209]: Can you please see if you can fix the indentation here?

File: cookbook/File_Upload.md [hunk 4]
```diff
@@ -433,20 +429,20 @@ We're already storing the attached image URL in state, so let's use the existenc
...
-<div
-  id="picker"
-  style={{
-    marginTop: '2rem',
-    height: '20rem',
-    display: url ? 'none' : 'block',
-  }}
-></div>
+>
+	<div
```
Comment (Tobbe) [inline cookbook/File_Upload.md:436]: Can you please fix the indentation here?

File: cookbook/File_Upload.md [hunk 5]
```diff
@@ -433,20 +429,20 @@ We're already storing the attached image URL in state, so let's use the existenc
...
+		id="picker"
+		style={{
+			marginTop: '2rem',
```
Comment (Tobbe) [inline cookbook/File_Upload.md:439]: Are all these styles still correct with the new inline picker?

File: cookbook/File_Upload.md [hunk 6]
```diff
@@ -138,7 +138,7 @@ const ImageForm = (props) => {
         />
         <FieldError name="title" className={CSS.errorMessage} />
 
-        <ReactFilestack apikey={process.env.REDWOOD_ENV_FILESTACK_API_KEY} />
+        <PickerInline apikey={process.env.REDWOOD_ENV_FILESTACK_API_KEY} />
```
Comment (Tobbe) [inline cookbook/File_Upload.md:141]: Aren't you missing a `<div>` as a child here? What does it look like if you render a `<PickerInline>` without that child?

**Review by Tobbe — CHANGES_REQUESTED** [review]

File: cookbook/File_Upload.md [hunk 7]
```diff
@@ -156,28 +156,11 @@ const ImageForm = (props) => {
 export default ImageForm
 ```
 
-If you look closely there's a *little* button under the Title input:
-
-![Pick file button](https://user-images.githubusercontent.com/300/82617171-1c3d7700-9b84-11ea-9e70-d005c419ebe1.png)
-
-Clicking that actually launches the picker with all kinds of options, like picking a local file, providing a URL or even grabbing one from Facebook, Instagram or Google Drive. Not bad!
+We now have created a picker with all kinds of options, like picking a local file, providing a URL or even grabbing one from Facebook, Instagram or Google Drive. Not bad!
```
Comment (Tobbe) [inline cookbook/File_Upload.md:159]:
```suggestion
We have now created a picker with all kinds of options, like picking a local file, providing a URL or even grabbing a file from Facebook, Instagram or Google Drive. Not bad!
```

File: cookbook/File_Upload.md [hunk 8]
```diff
@@ -156,28 +156,11 @@ const ImageForm = (props) => {
...
-<ReactFilestack
-  apikey={process.env.REDWOOD_ENV_FILESTACK_API_KEY}
-  componentDisplayMode={{ type: 'immediate' }}
-  actionOptions={{ displayMode: 'inline', container: 'picker' }}
-/>
-<div id="picker" style={{ marginTop: '2rem', height: '20rem' }}></div>
-```
-
-Great! You can even try uploading an image to make sure it works:
+... And....you can even try uploading an image to make sure it works:
```
Comment (Tobbe) [inline cookbook/File_Upload.md:163]:
```suggestion
You can even try uploading an image to make sure it works:
```

File: cookbook/File_Upload.md [hunk 9]
```diff
@@ -430,23 +410,22 @@ It would be nice if, after uploading, you could see the image you uploaded. Like
 
 We're already storing the attached image URL in state, so let's use the existence of that state to show the attached image. In fact, let's also hide the uploader and assume you're done (you'll be able to show it again if needed):
 
+For this we will need define a custom `<div>` and have it as a child of `<PickerInline>`. We'll also have the `<div>` display or hide itself (ie. the picker), if there exists a defined image url:
```
Comment (Tobbe) [inline cookbook/File_Upload.md:413]:
```suggestion
For this we will need to define a custom `<div>` and make it a child of `<PickerInline>`. We'll also have the `<div>` display or hide itself (i.e. the picker), if there exists a defined image url:
```

**Comment by Tobbe** [comment]
This is starting to look really good @vchoy! I'm going to try to follow the updated cookbook recipe and see if any more changes are needed. Please remind me if you haven't heard back in a few days.

File: cookbook/File_Upload.md [hunk 10]
```diff
@@ -430,23 +410,22 @@ It would be nice if, after uploading, you could see the image you uploaded. Like
 
 We're already storing the attached image URL in state, so let's use the existence of that state to show the attached image. In fact, let's also hide the uploader and assume you're done (you'll be able to show it again if needed):
 
+For this we will need to define a custom `<div>` and make it a child of `<PickerInline>`. We'll also have the `<div>` display or hide itself (i.e. the picker), if there exists a defined image url:
+
 ```javascript{14,18}
```
Comment (Tobbe) [inline cookbook/File_Upload.md:415]: You need to update the highlighted lines by changing the numbers here (14,18 should be something else)
Also, try changing from javascript to jsx, see if that makes the highlighting look better

File: cookbook/File_Upload.md [hunk 11]
```diff
@@ -430,23 +410,22 @@ It would be nice if, after uploading, you could see the image you uploaded. Like
...
-    height: '20rem',
-    display: url ? 'none' : 'block',
-  }}
-></div>
+>
+  <div
+    style={{
+      display: url ? 'none' : 'block',
+    }}
+  ></div>
```
Comment (Tobbe) [inline cookbook/File_Upload.md:426]:
```suggestion
  <div style={{ display: url ? 'none' : 'block' }}></div>
```

Comment (Tobbe) [reply cookbook/File_Upload.md:426]: I'm not seeing the file picker. Please fix the code so it's possible to upload an image

image

File: cookbook/File_Upload.md [hunk 12]
```diff
@@ -460,20 +439,16 @@ Now let's just add the ability to bring back the uploader if you decide you want
...
+</PickerInline>
 
 {url && (
```
Comment (Tobbe) [inline cookbook/File_Upload.md:453]:
```jsx
        {url && (
          <div>
            <img src={url} style={{ display: 'block', marginTop: '2rem 0' }} />
            <button
              onClick={() => setUrl(null)}
              className="rw-button rw-button-blue"
            >
              Replace image
            </button>
          </div>
        )}
```

File: cookbook/File_Upload.md [hunk 13]
```diff
@@ -460,20 +439,16 @@ Now let's just add the ability to bring back the uploader if you decide you want
 ```javascript{18-29}
```
Comment (Tobbe) [inline cookbook/File_Upload.md:439]: these line numbers are wrong

**Review by Tobbe — CHANGES_REQUESTED** [review]

**Comment by thedavidprice** [comment]
Hi gents! Just checking in on this PR -- it's looking good but not yet complete, correct?

**Comment by Tobbe** [comment]
@thedavidprice correct!

**Pull request merged**